### ISSUE TITLE
Pull requests aren't synced until page refresh

### ISSUE THREAD

**Issue by seanlinsley** [huboard/huboard#677]
This all happened in our Slack room:

<img width="583" alt="screen shot 2016-06-13 at 10 32 05 am" src="https://cloud.githubusercontent.com/assets/688886/16013662/f0cd24d8-3153-11e6-9214-cccd071bda8c.png">

But the PR (it was a PR without a ticket) stayed in Review, and would only move over after performing a page refresh.

**Comment by seanlinsley** [comment]
Is this a duplicate of #668?

**Comment by dahlbyk** [comment]
This shouldn't depend on #668, no.

A naive test of this scenario with https://github.com/huboard-testing/spooky-octo-pug/pull/99 did successfully move the card into Done for me on merge. Anything else you can tell me about your environment/repo that might help us narrow in on the cause here?

**Comment by seanlinsley** [comment]
It looks like it was moved before it was merged / closed. I don't know if it was moved via Huboard or by manually updating the label in Github. @thepug do you remember?

<img width="465" alt="screen shot 2016-06-13 at 12 38 19 pm" src="https://cloud.githubusercontent.com/assets/688886/16017053/c54cf3d2-3163-11e6-98ac-a82f5bbb4561.png">

**Comment by thepug** [comment]
I moved the item in huboard

**Comment by dahlbyk** [comment]
So to confirm, this is what were expecting to see "3 hours ago" on merge, but it didn't happen:
![pr-move](https://cloud.githubusercontent.com/assets/133987/16017823/3dd6b5ce-3167-11e6-8f96-c8f7e69c299a.gif)

But you did get the "PR closed" notification in Slack. Can you confirm that the settings Health Check shows that all three web hooks are configured?

**Comment by seanlinsley** [comment]
Yep, the health check passes.

**Comment by dahlbyk** [comment]
I'm kind of at a loss here... This happens consistently for all your PR close/merge actions? We might need to share a screen or something to help troubleshoot, as I can't reproduce what you describe.

**Comment by seanlinsley** [comment]
I'd be happy to screenshare with you. I could also show you another issue I've noticed: immediately after commenting on a ticket, it's common for that ticket to disappear from Huboard.

**Comment by seanlinsley** [comment]
I've also seen ticket disappearance happen when modifying the ticket description.

And, mysteriously, two tickets today have disappeared while I was moving them to a different column. That is: after releasing my mouse, the ticket can no longer be found on the page.

**Comment by discorick** [comment]
Uh oh, thats some really concerning behavior! Did the cards "reappear" in the destination column on refresh ?

**Comment by seanlinsley** [comment]
IIRC, they had to be moved again.

It seems pretty likely that the recently-added resync feature (or changes around it) are causing these bugs.

**Comment by discorick** [comment]
It _could_ be a combination of factors interacting with the resync.. are you using Firefox by any chance @seanlinsley ? I'm looking into some strange caching behavior on FF that can really throw HB into bad states even after a refresh.

**Comment by seanlinsley** [comment]
@discorick yes I'm using Firefox.

**Comment by discorick** [comment]
@dahlbyk asked me to post my request headers here, this is for the ajax call on board load:

```
Host: huboard.rails
User-Agent: Mozilla/5.0 (Macintosh; Intel Mac OS X 10.10; rv:47.0) Gecko/20100101 Firefox/47.0
Accept: */*
Accept-Language: en-US,en;q=0.5
Accept-Encoding: gzip, deflate
X-Requested-With: XMLHttpRequest
Referer: http://huboard.rails/discorick/skipping.stones
Cookie: ajs_user_id=number; ajs_group_id=null; ajs_anonymous_id=number; _huboard-web_session=number
Connection: keep-alive
If-None-Match: W/"5539ddd8bc201de471ab541f02136b83"
Cache-Control: max-age=0
```

**Comment by dahlbyk** [comment]
Adding `no-cache` couldn't hurt, though most browsers interpret `max-age=0` equivalently.

**Comment by discorick** [comment]
I just opened up https://github.com/huboard/huboard-web/pull/326 which I suspect will have a positive effect on these issues - Firefox sometimes uses cached ajax responses to hydrate the board data which makes the board interaction really unpredictable.
